src/mono/mono/mini/aot-compiler.c

```diff
@@ -8513,6 +8541,24 @@ mono_aot_parse_options (const char *aot_options, MonoAotOptions *opts)
 			mini_debug_options.no_seq_points_compact_data = FALSE;
 			opts->gen_msym_dir = TRUE;
 			opts->gen_msym_dir_path = g_strdup (arg + strlen ("msym_dir="));
+		} else if (str_begins_with (arg, "direct-pinvokes=")) {
+			char *direct_pinvokes = g_strdup (arg + strlen ("direct-pinvokes="));
+			gchar *direct_pinvoke_ctx = NULL;
+			gchar *direct_pinvoke = strtok_r (direct_pinvokes, ",", direct_pinvoke_ctx);
```

Comment:
Maybe "," is not a good delimitator since its already used to split different aot arguments apart like:

--aot=full,soft-debug,direct-pinvokes=

so that probably enforce use of " if you have multiple values. Alternative switch to a different delimitator that can not be used as part of module, symbol or file names or follow pattern used by other arguments that can be multi value like you had before, and just adjust the name of the argument indicate that it carries one value and a not a collection of values.

Reply:
Good catch, changed the separator to be a semicolon instead.